packages/web-frontend/cypress/scripts/testE2e/testE2e.js

```diff
@@ -5,41 +5,109 @@
 
 import { execSync } from 'child_process';
 import {} from 'dotenv/config';
+import fs from 'fs';
 
 import { getArgs, getLoggerInstance } from '@tupaia/utils';
+import { E2E_CONFIG_PATH } from '../../constants';
+import { generateConfig } from '../generateConfig/generateConfig';
 
 const scriptConfig = {
   options: {
     ciBuildId: {
       type: 'string',
       description:
-        'Used to associate multiple CI machines to one test run, see https://docs.cypress.io/guides/guides/parallelization#Linking-CI-machines-for-parallelization-or-grouping',
+        'Used in CI/CD builds to associate multiple CI machines with one test run, see https://docs.cypress.io/guides/guides/parallelization#Linking-CI-machines-for-parallelization-or-grouping',
     },
   },
 };
 
-const runPackageScript = script =>
-  execSync(`yarn workspace @tupaia/web-frontend ${script}`, { stdio: 'inherit' });
+const printResults = ({ baseUrl, compareUrl }, { baseError, compareError }) => {
+  const logger = getLoggerInstance();
+
+  logger.info();
+
+  const printResult = (description, error) => {
+    if (error) {
+      logger.error(`❌ ${description}`);
```

Comment:
Do we want to log the error as well?

Reply:
We do! It will have been logged already (in https://github.com/beyondessential/tupaia/blob/e337ee627a52360701cc8a0ee2b0e6a203d5fe80/packages/web-frontend/cypress/scripts/testE2e/testE2e.js#L98)

This is just to show a summary of the results per url in the end:

![image](https://user-images.githubusercontent.com/20692464/125705144-4ca01426-0355-4a49-9ef7-42982d0401e6.png)

Reply:
Gotcha, great!